src/libmongoc/src/mongoc/mongoc-cluster-aws.c

```diff
@@ -432,6 +488,214 @@ _obtain_creds_from_ecs (_mongoc_aws_credentials_t *creds, bson_error_t *error)
    return ret;
 }
 
+static bool
+_obtain_creds_from_assumerolewithwebidentity (_mongoc_aws_credentials_t *creds,
+                                              bson_error_t *error)
+{
+   bool ret = false;
+   char *http_response_headers = NULL;
+   char *http_response_body = NULL;
+   char *AWS_WEB_IDENTITY_TOKEN_FILE = NULL;
+   char *AWS_ROLE_ARN = NULL;
+   char *AWS_ROLE_SESSION_NAME = NULL;
```

Comment:
```suggestion
   char *aws_web_identity_token_file = NULL;
   char *aws_role_arn = NULL;
   char *aws_role_session_name = NULL;
```
Suggest avoiding full `UPPER_CASE` for non-macro entities.